solr/solrj/src/java/org/apache/solr/client/solrj/impl/CloudHttp2SolrClient.java

```diff
@@ -332,6 +335,11 @@ public Builder withRetryExpiryTime(long expiryTime, TimeUnit unit) {
       return this;
     }
 
+    /** Sets the default collection for request. */
+    public Builder withDefaultCollection(String collection) {
```

Comment:
Would you like to push up an example of what you are thinking?   I'm not wedded to how this works ;-).

Reply:
(done)

Reply:
I'm not sure I see the benefit of throwing an error when a different collection is used than the default. I feel like the idea behind this option is just to make it easier for people to not have to specify the collection in their commands most of the time. If you are explicitly setting a collection in a command, its pretty unlikely that that was an error...

Reply:
Right; but nearby, easy to **not** specify a collection in a command and **not** communicate with the intended collection.  For example imagine some code that looks like:
```
addSomeDocs(collectionName)
....
commit()
```
My real wish/preference isn't reflected in this PR at all actually, which is for there to be a collection scoped thing that you can interact with, instead of many overloads that are optional.  Maybe let's keep defaultCollection as that and then some future day work towards that (if I can convince anyone).

Reply:
i think the whole `collectionName` capablity is maybe a bad idea....  It makes the CloudSolrClient and regular old SolrClient very different behaviors...